Ammp 1269 improvement mqtt schema

judruizc opened the pull request:
Hi @svet-b, please have a look to this PR. 
changes here are mainly: naming, adding vendor_id and time in unix epoch (API payload still uses date-like format)
I have tested locally and I'm getting this as device-based readout (mqtt):

```
{
  "t": 1606918060,
  "r": [
    {
      "_d": "logger",
      "_vid": "3c:22:fb:63:d6:f3",
      "boot_time": 1606464128,
      "cpu_load": 3.142578125,
      "memory_usage": 66.3
    }
  ],
  "m": {
    "snap_rev": 0,
    "config_id": "43b9de6",
    "reading_duration": 0.006443,
    "reading_offset": 0
  }
}
```

And this for the API payload:
```
{
  "time": "2020-12-02T15:07:40Z",
  "fields": {
    "comms_lggr_boot_time": 1606464128,
    "comms_lggr_cpu_load": 3.142578125,
    "comms_lggr_mem_usage": 66.3,
    "snap_rev": 0,
    "config_id": "43b9de6",
    "reading_duration": 0.006443,
    "reading_offset": 0
  }
}
```

Let me know what can be improved !

svet-b reviewed (changes requested):
Thanks! It looks like it pretty much does what it needs! I've added a few suggestions for potential improvement.

judruizc commented:
Hey @svet-b, thank you for your advice. I think I've implemented all of them. 
I also realised that the Influx endpoint was most likely going to fail as it was using the device-based readout. I think I fixed that too, but not sure how to test that endpoint (and if is still relevant). 

Happy to keep improving if needed :)

svet-b commented:
Thanks @judruizc, looks good!
I've made a couple of updates:
- The function for converting to API payload needed to be refactored. It didn't actually correctly treat a situation where multiple devices of the same driver were returning data into the same variables. It would have only looked up data based on variable name, and ignored the device from which the data came. Also I made it create a new payload from scratch rather than "reshaping" the existing one, which I think was quite confusing.
- Unified the way that 'reading_offset' is treated. Depending on where the data was going, we were either putting it inside or outside of the metadata object in the device-based reading, and then doing something funny with it in the API payload converter. This is a bit more straightforward I think.
- Some formatting, mostly to avoid long line lengths and satisfy Flake8 as much as possible.

I haven't had the chance to test this properly, but will try to do that and then we can merge.

svet-b commented:
Just pushed a fix for a bug that I introduced last night. Seems to be working now. Can you test? If it looks good we can push to dev, and later today maybe also merge with master in order to deploy this to a datalogger
